Ignore selection of invalid model when editing layout

leighfiddes opened the pull request:
When adding or deleting items from the Layout, OnSelectionChanged() will
be called pointing to invalid models which was causing an assert.
Change to log and ignore.
This appears to be a similar problem to that described in wxWidgets ticket #18045.
This addresses one item in Issue #1933.
@cjd please consider.

keithsw1111 commented:
I would be more interested in a fix which identified why we were holding onto an invalid model pointer. Also this would not be a debug message but a error or critical message.
If you can reliably reproduce it maybe you could post a video of that.

leighfiddes commented:
Thanks for your response Keith. I have captured videos and log of the issue. I'm using 2020.11 on Kubuntu 18.04 LTS.
1) Adding a model group causes immediate crash. See video "add.mp4"
2) Deleting a model causes immediate crash. See video "delete.mp4"
I then built a debug build of 2020.11
3) Deleting a model causes an Assert message. See video "delete_debug.mp4"
I added my proposed patch. I have also added extra debugging (prefixed by "LF")
4) Deleting a model seems to work fine. See video "delete_patched.mp4". I have also extracted the relevant lines from "/tmp/xLights_l4cpp.log" in the file "log_extract.txt". You can see after the DELETE_MODEL_GROUP that the old models are still being referenced, for example "Arches" was at 0x557cd079e0f0 before the deletion but this pointer now does not point to a valid model. "Arches" has moved to 0x557cd0a67ee0.
[videos.zip](https://github.com/smeighan/xLights/files/4379866/videos.zip)

keithsw1111 commented:
Do you have your show folder ... I wonder if this isnt related to the leading trailing space changes I have been trying to make.

leighfiddes commented:
This is the simple show I used for the "delete_debug" and "delete_patched" videos.
[testshow.zip](https://github.com/smeighan/xLights/files/4386555/testshow.zip)

keithsw1111 commented:
So I added groups, deleted models ... no crashes

leighfiddes commented:
I thought you only ran Windows. Do you run Linux as well?

keithsw1111 commented:
I run windows ... if the bug is linux specific then it is likely event related. I still need to find the cause ... not keen to just treat symptoms

leighfiddes commented:
I appreciate you efforts Keith. To me it appears to be a mismatch with the wxGTK interface library which behaves differently to the WxWindows library because obviously the Windows version doesn't crash. As I initially noted, this appears similar but not identical to the issue in [http://trac.wxwidgets.org/ticket/18045](url).

keithsw1111 commented:
I have included this but in a way where it only runs on Linux as I am led to believe this is purely a linux issue,
